File: SoundJavaSample/app/src/main/java/com/example/soundsample/MainActivity.java
package com.example.soundsample;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        final MainSoundPlayer soundPlayer = new MainSoundPlayer(this);
        getLifecycle().addObserver(soundPlayer);
        // 効果音の再生/停止処理
        findViewById(R.id.btnCurtainSoundPlay).setOnClickListener(v -> soundPlayer.playCurtainSound());
        findViewById(R.id.btnCurtainSoundStop).setOnClickListener(v -> soundPlayer.stopCurtainSound());

        findViewById(R.id.btnWindowoundPlay).setOnClickListener(v -> soundPlayer.playWindowSound());
        findViewById(R.id.btnWindowSoundStop).setOnClickListener(v -> soundPlayer.stopWindowSound());
    }
}

File: SoundKotlinSample/app/src/main/java/com/example/soundsample/MainActivity.kt
package com.example.soundsample

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val soundPlayer = MainSoundPlayer(this)
        lifecycle.addObserver(soundPlayer)
        // 効果音の再生/停止処理
        findViewById<View>(R.id.btnCurtainSoundPlay).setOnClickListener { soundPlayer.playCurtainSound() }
        findViewById<View>(R.id.btnCurtainSoundStop).setOnClickListener { soundPlayer.stopCurtainSound() }
        findViewById<View>(R.id.btnWindowoundPlay).setOnClickListener { soundPlayer.playWindowSound() }
        findViewById<View>(R.id.btnWindowSoundStop).setOnClickListener { soundPlayer.stopWindowSound() }
    }
}